lib/vehicle_factory.rb

```diff
@@ -0,0 +1,13 @@
+class VehicleFactory
+  def initialize
+    @factory_vehicles = []
+  end
+
+  def create_vehicles(api_data)
+    api_data.each do |data|
+      vehicle = Vehicle.new(data)
+      @factory_vehicles << vehicle
+    end
+    @factory_vehicles
+  end
+end
```

Comment:
Great job here. I see on line `11` is returning the `@factory_vehicles` array at the final operation in the method. You can eliminate the need to do this if you switch out `each` for `map` like so:

```ruby
require "./lib/vehicle"

class VehicleFactory
  def create_vehicles(api_data)
    api_data.map do |data|
      Vehicle.new(data)
    end
  end
end
```

1. You'll see I completely got rid of the `initialize` and the `@factory_vehicles` array altogether
2. Converted `each` to `map` since `map` transforms every element in an array from one thing to something else, in this case -- `Vehicle` objects
3. Don't forget to require the `Vehicle` class at the top of this file since you are instantiating new vehicles